lib/packages/core/lib/src/views/media/media_preview.dart

```diff
@@ -0,0 +1,280 @@
+import 'package:flutter/material.dart';
+import 'package:likeminds_chat_flutter_core/src/core/core.dart';
+import 'package:likeminds_chat_flutter_core/src/convertors/convertors.dart';
+import 'package:likeminds_chat_flutter_core/src/utils/utils.dart';
+import 'package:likeminds_chat_flutter_core/src/views/media/configurations/preview/builder.dart';
+import 'package:likeminds_chat_flutter_ui/likeminds_chat_flutter_ui.dart';
+
+/// {@template lm_chat_media_preview_screen}
+/// A screen to preview media before adding attachments to a conversation
+///
+/// Creates a new instance for [LMChatMediaPreviewScreen]
+///
+/// Gives access to customizations through instance builder variables
+///
+/// To configure the page, use [LMChatMediaForwardingScreenConfig]
+/// {@endtemplate}
+class LMChatMediaPreviewScreen extends StatefulWidget {
+  /// required conversation object to fetch media and title
+  final LMChatConversationViewData? conversation;
+
+  /// {@macro lm_chat_media_preview_screen}
+  const LMChatMediaPreviewScreen({
+    super.key,
+    this.conversation,
+  });
+
+  @override
+  State<LMChatMediaPreviewScreen> createState() =>
+      _LMChatMediaPreviewScreenState();
+}
+
+class _LMChatMediaPreviewScreenState extends State<LMChatMediaPreviewScreen> {
+  int currPosition = 0;
+  List<LMChatMediaModel> mediaList = [];
+
+  ValueNotifier<bool> rebuildCurr = ValueNotifier<bool>(false);
+
+  final LMChatMediaPreviewBuilderDelegate _screenBuilder =
+      LMChatCore.config.mediaPreviewConfig.builder;
+
+  @override
+  void initState() {
+    super.initState();
+    mediaList = LMChatMediaHandler.instance.pickedMedia;
+  }
+
+  @override
+  void didUpdateWidget(covariant LMChatMediaPreviewScreen oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    mediaList = LMChatMediaHandler.instance.pickedMedia;
+  }
+
+  @override
+  void deactivate() {
```

Comment:
Shouldn’t we have used `dispose` here?

Reply:
No dispose is used when it is removed from the tree, we need the lifecycle where it might still be in the tree but is now deactivated.